modules/platform/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/BridgeCommentManager.java

```diff
@@ -160,28 +168,34 @@ public Comment getComment(CoreSession session, String commentId)
     public Comment updateComment(CoreSession session, String commentId, Comment comment)
             throws CommentNotFoundException, CommentSecurityException {
         DocumentRef commentRef = new IdRef(commentId);
-        if (!session.exists(commentRef)) {
-            throw new CommentNotFoundException("The comment " + commentId + " does not exist");
-        }
-        if (session.getDocument(commentRef).getPropertyValue(COMMENT_PARENT_ID_PROPERTY) != null) {
-            return second.updateComment(session, commentId, comment);
-        } else {
-            return first.updateComment(session, commentId, comment);
-        }
+        return CoreInstance.doPrivileged(session, s -> {
+            // retrieve comment to check which service handles it
+            if (!s.exists(commentRef)) {
+                throw new CommentNotFoundException("The comment " + commentId + " does not exist");
+            }
+            if (s.getDocument(commentRef).getPropertyValue(COMMENT_PARENT_ID_PROPERTY) != null) {
+                return second.updateComment(session, commentId, comment);
+            } else {
+                return first.updateComment(session, commentId, comment);
+            }
+        });
     }
 
     @Override
     public void deleteComment(CoreSession session, String commentId)
             throws CommentNotFoundException, CommentSecurityException {
         DocumentRef commentRef = new IdRef(commentId);
-        if (!session.exists(commentRef)) {
-            throw new CommentNotFoundException("The comment " + commentId + " does not exist");
-        }
-        if (session.getDocument(commentRef).getPropertyValue(COMMENT_PARENT_ID_PROPERTY) != null) {
-            second.deleteComment(session, commentId);
-        } else {
-            first.deleteComment(session, commentId);
-        }
+        CoreInstance.doPrivileged(session, s -> {
+            // retrieve comment to check which service handles it
+            if (!s.exists(commentRef)) {
+                throw new CommentNotFoundException("The comment " + commentId + " does not exist");
+            }
+            if (s.getDocument(commentRef).getPropertyValue(COMMENT_PARENT_ID_PROPERTY) != null) {
+                second.deleteComment(session, commentId);
+            } else {
+                first.deleteComment(session, commentId);
+            }
+        });
```

Comment:
same as above where we can replace with:
```
 @Override
    public void deleteComment(CoreSession session, String commentId)
            throws CommentNotFoundException, CommentSecurityException {
        DocumentRef commentRef = new IdRef(commentId);
        execute(session, commentRef, cm -> {
            cm.deleteComment(session, commentId);
            return true;
        });
    }
```
here I just wrap the deletion to return a boolean because the execute `function` param is function Type

Reply:
Same as other.